R/db.R

```diff
@@ -284,11 +284,11 @@ write_data_safely <- function(dbconn, tblname, df, key, safe_columns, blacklist)
     if (length(blacklist) > 0L) {
       df_without_id <- df[-which(names(df) == key)]  # Don't blacklist on id.
       if (any(is.na(blacklist))) {
-        df <- df[apply(!is.na(df_without_id), 1, all),]
+        df <- df[apply(!is.na(df_without_id), 1, all), , drop = FALSE]
         blacklist <- blacklist[!is.na(blacklist)]
       }
       if (length(blacklist) > 0L) {
-        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))),]
+        df <- df[apply(df_without_id, 1, function(x) all(!(x %in% blacklist))), , drop = FALSE]
```

Comment:
@robertzk I think there's a cool opportunity here for making this more DRY and using function composition to create a `filter_fn` and for the `apply(df_without_id, 1, filter_fn)`